contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,949 @@
+    event Delegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Undelegated(address indexed operatorAddress, address indexed delegator, uint256 shares, uint256 bnbAmount);
+    event Redelegated(
+        address indexed srcValidator,
+        address indexed dstValidator,
+        address indexed delegator,
+        uint256 oldShares,
+        uint256 newShares,
+        uint256 bnbAmount
+    );
+    event RewardDistributed(address indexed operatorAddress, uint256 reward);
+    event RewardDistributeFailed(address indexed operatorAddress, bytes failReason);
+    event ValidatorSlashed(
+        address indexed operatorAddress, uint256 jailUntil, uint256 slashAmount, SlashType slashType
+    );
+    event ValidatorJailed(address indexed operatorAddress);
+    event ValidatorEmptyJailed(address indexed operatorAddress);
+    event ValidatorUnjailed(address indexed operatorAddress);
+    event Claimed(address indexed operatorAddress, address indexed delegator, uint256 bnbAmount);
+    event Paused();
+    event Resumed();
+
+    /*----------------- modifiers -----------------*/
+    modifier validatorExist(address operatorAddress) {
+        if (!_validatorSet.contains(operatorAddress)) revert ValidatorNotExist();
+        _;
+    }
+
+    modifier whenNotPaused() {
+        if (_paused) revert StakeHubPaused();
+        _;
+    }
+
+    modifier onlyAssetProtector() {
+        if (msg.sender != assetProtector) revert OnlyAssetProtector();
+        _;
+    }
+
+    modifier notInBlackList() {
+        if (blackList[msg.sender]) revert InBlackList();
+        _;
+    }
+
+    receive() external payable {
+        // to prevent BNB from being lost
+        if (_isRedelegating != 1) revert();
+    }
+
+    /**
+     * @dev this function is invoked by BSC Parlia consensus engine during the hard fork
+     */
+    function initialize() external initializer onlyCoinbase onlyZeroGasPrice {
+        transferGasLimit = 5000;
+        minSelfDelegationBNB = 2_000 ether;
+        minDelegationBNBChange = 1 ether;
+        maxElectedValidators = 29;
+        unbondPeriod = 7 days;
+        redelegateFeeRate = 2;
+        downtimeSlashAmount = 10 ether;
+        felonySlashAmount = 200 ether;
+        downtimeJailTime = 2 days;
+        felonyJailTime = 30 days;
+        felonyPerDay = 2;
+
+        address[] memory bcConsensusAddress;
+        bytes[] memory bcVoteAddress;
+        bcConsensusAddress = abi.decode(INIT_BC_CONSENSUS_ADDRESSES, (address[]));
+        bcVoteAddress = abi.decode(INIT_BC_VOTE_ADDRESSES, (bytes[]));
+        for (uint256 i; i < bcConsensusAddress.length; ++i) {
+            _legacyConsensusAddress[bcConsensusAddress[i]] = true;
+        }
+        for (uint256 i; i < bcVoteAddress.length; ++i) {
+            _legacyVoteAddress[bcVoteAddress[i]] = true;
+        }
+
+        assetProtector = 0xdF87F0e2B8519Ea2DD4aBd8B639cdD628497eD25; // TODO
```

Comment:
Please use a dead address in the contract

Reply:
I hope it can be a dead address in the code, you can change it through gov or before the testcase begin.

Reply:
Done